activerecord/lib/active_record/railtie.rb

```diff
@@ -276,5 +276,13 @@ class Railtie < Rails::Railtie # :nodoc:
         self.signed_id_verifier_secret ||= -> { Rails.application.key_generator.generate_key("active_record/signed_id") }
       end
     end
+
+    initializer "active_record.query_comment" do
+      ActiveSupport.on_load(:active_record) do
+        name = Rails.application.class.name.split("::").first
+        ConnectionAdapters::AbstractAdapter::QueryCommentContext.update({application_name: name})
+        ConnectionAdapters::AbstractAdapter::QueryCommentContext.backtrace_cleaner = ::Rails.backtrace_cleaner
```

Comment:
I think the `::` isn't required here, other Rails callers in this file aren't adding it.

```suggestion
        ConnectionAdapters::AbstractAdapter::QueryCommentContext.backtrace_cleaner = Rails.backtrace_cleaner
```